src/m_menu.c

```diff
@@ -6868,11 +6919,37 @@ void M_InitHelpScreen()
     }
 }
 
+static void M_GetMidiDevices(void)
+{
+#if defined(_WIN32)
+  int i;
+  extern int I_WIN_DeviceList(const char *devices[], int max_devices);
+
+  static const char *s[] = {
+  #if defined(HAVE_FLUIDSYNTH)
+    "FluidSynth",
+  #endif
+    "OPL Emulation",
+    NULL
+  };
+
+  winmm_num_devices = I_WIN_DeviceList(midi_player_strings,
+                                       MAX_MIDI_PLAYERS - arrlen(s));
+
+  for (i = 0; i < arrlen(s); ++i)
+  {
+    midi_player_strings[winmm_num_devices + i] = s[i];
```

Comment:
We could run `M_StringWidth()` over all device names, determine the maximum value and use this to dynamically adjust the indentation value in `gen_settings1[]` which is currently hard-coded to `150`.

Reply:
The maximum length of a Windows MIDI device name is 32 (thanks to Ceski):
```c
/* general constants */
#define MAXPNAMELEN      32     /* max product name length (including NULL) */
```
The length of the "Microsoft GS Wavetable Synth" that everyone has is 28. With `150` a 32-character string fits a 4:3 screen, so I think that's fine.

Reply:
Alright, thanks!